docs/enhancements/patch-install-manifests.md

```diff
@@ -0,0 +1,203 @@
+`openshift-install create manifests` prior to feeding them to `openshift-install create cluster`.
+Moving forward, installer frequently prefers new features to be enabled via this latter mechanism.
+
+Up to this point, hive has incorporated some piecemeal patching of these manifests at the behest of
+specific values passed through via hive APIs.
+Example: Extra worker security groups for AWS.
+However, this model is awkward, brittle, and not extensible.
+What is needed is a mechanism for the hive *user* to specify arbitrary modifications to OpenShift
+manifests generated by the installer.
+
+## Groundwork
+
+### Existing Manifest Patching
+The code that invokes `openshift-install` lives in installmanager.go.
+[This block](https://github.com/openshift/hive/blob/9615c77cd786dd82079a9dacd60b64c882f56327/pkg/installmanager/installmanager.go#L882-L921)
+contains logic to:
+- Walk the `openshift/` directory (one of two created by `openshift-install create manifests`, the
+  other being `manifests/`).
+- Load yaml files.
+- Parse the files as JSON.
+- Patch the JSON.
+- Convert the JSON back to YAML.
+- Write the files back to storage.
+
+### `ClusterDeploymentCustomization`
+An earlier feature, [ClusterPool Inventory](clusterpool-inventory.md), invented the concept of
+[`ClusterDeploymentCustomization`](https://github.com/openshift/hive/blob/8c79fedae7011e434e8a7ff0fef40994ef92deb2/vendor/github.com/openshift/hive/apis/hive/v1/clusterdeploymentcustomization_types.go),
+a hive API used by ClusterPools when specific discrete values (such as reserved IP addresses) must
+be injected into the install-config.yaml for each cluster in the pool.
+As such, ClusterDeploymentCustomization originally contained only one member,
+[`InstallConfigPatches`](https://github.com/openshift/hive/blob/4eaf5fd7858e8def3b2ba3229801090610246243/vendor/github.com/openshift/hive/apis/hive/v1/clusterdeploymentcustomization_types.go#L45),
+a list of patches to apply to the install-config.yaml generated by the ClusterPool controller based
+on the user-provided template.
+Each entry in `InstallConfigPatches` is a
+[`PatchEntity`](https://github.com/openshift/hive/blob/4eaf5fd7858e8def3b2ba3229801090610246243/vendor/github.com/openshift/hive/apis/hive/v1/clusterdeploymentcustomization_types.go#L49),
+a hive-owned CRD corresponding to an
+[RFC 6902 JSON patch](https://datatracker.ietf.org/doc/html/rfc6902).
+
+## API
+
+### `ClusterDeploymentCustomization.Spec.InstallerManifestPatches`
+
+**Example:**
+```yaml
+apiVersion: v1
+kind: ClusterDeploymentCustomization
+metadata:
+  name: foo-cluster-deployment-customization
+  namespace: my-project
+spec:
+  installerManifestPatches:
+    # Add custom labels to all master machine manifests
+    - manifestSelector:
+        glob: openshift/99_openshift-cluster-api_master-machines-*.yaml
+      patches:
+        - op: add
+          path: metadata/labels/a-custom-label
```

Comment:
path should be "/metadata/labels/a-custom-label".
Need '/' at the beginning of the path otherwise jsonpatch reports the following error:
add operation does not apply: doc is missing path: \"metadata/labels/a-custom-label\"